dspy/predict/predict.py

```diff
@@ -247,14 +254,21 @@ def new_generate(lm, signature, example, max_depth=6, **kwargs):
 
 def v2_5_generate(lm, lm_kwargs, signature, demos, inputs, _parse_values=True):
     import dspy
-    adapter = dspy.settings.adapter or dspy.ChatAdapter()
 
-    return adapter(lm, lm_kwargs=lm_kwargs, signature=signature, demos=demos, inputs=inputs, _parse_values=_parse_values)
-    
+    adapter = dspy.settings.adapter or dspy.ChatAdapter()
+    return adapter(
+        lm,
+        lm_kwargs=lm_kwargs,
+        signature=signature,
+        demos=demos,
+        inputs=inputs,
+        _parse_values=_parse_values,
+        max_extensions=dspy.settings.max_extensions or 0,
```

Comment:
@okhat this is the only real change in this file - everything else is linting/formatting from rust